src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -295,6 +305,44 @@ double Camera::extract_near_z() const
     return near_z;
 }
 
+namespace
+{   
+    float map_sample_to_linear_curve(const float a, const float b, const float x)
+    {
+        float constant;
+        // opening
+        if (a > 0)
+        {
+            constant = 0;
+        }
+        // closing
+        else
+        {
+            constant = 1 - (a / 2) - b;
+        }
+        return (sqrt(2 * a * (x - constant) + b * b) - b) / a;
```

Comment:
I couldn't optimize better then this. :(

Reply:
Well this looks pretty good already!